cloudesire-api-client/src/main/java/com/cloudesire/platform/apiclient/api/CouponApi.java

```diff
@@ -25,30 +26,44 @@
     Call<Void> delete( @Path( "id" ) Integer id );
 
     @POST( "coupon" )
-    Call<CouponDTO> create( @Query( "type" ) String type, @Query( "expiration" ) Date expiration,
-            @Query( "value" ) BigDecimal value, @Query( "code" ) String code,
+    Call<List<CouponDTO>> generateList( @Query( "howMany" ) Integer howMany, @Query( "type" ) String type,
             @Query( "productVersion" ) Integer productVersion, @Query( "product" ) Integer product,
-            @Query( "licenseOnly" ) Boolean licenseOnly, @Query( "reusable" ) Boolean reusable );
+            @Query( "expirationDate" ) ISO8601DateTime expiration, @Query( "licenseOnly" ) Boolean licenseOnly,
+            @Query( "number" ) BigDecimal number );
 
-    @GET( "coupon/{id}" )
-    Call<CouponDTO> get( @Path( "id" ) Integer id );
+    @POST( "coupon" )
+    Call<CouponDTO> generate( @Query( "type" ) String type, @Query( "productVersion" ) Integer productVersion,
+            @Query( "product" ) Integer product, @Query( "expirationDate" ) ISO8601DateTime expiration,
+            @Query( "days" ) Integer days, @Query( "plafond" ) BigDecimal plafond );
+
+    @POST( "coupon" )
+    Call<CouponDTO> generate( @Query( "type" ) String type, @Query( "code" ) String code,
+            @Query( "productVersion" ) Integer productVersion, @Query( "product" ) Integer product,
+            @Query( "expiration" ) ISO8601DateTime expiration, @Query( "licenseOnly" ) Boolean licenseOnly,
+            @Query( "value" ) BigDecimal value );
```

Comment:
try to use the same parameter ordering for all the 3 generate methods (howMany can be the last parameter to differentiate between one or many coupon generation)

there is no need to call generateList, method signature already says that